getstorybook shouldn't use yarn if there's no `yarn.lock`

### Bug(?)

Using latest `@storybook/cli` running `getstorybook` installs alpha version of storybook versus latest stable version.

### Steps to reproduce

```bash
npm i -g @storybook/cli
getstorybook
```

The installed version of Storybook dependencies are `alpha` versions. Considering the versions are alpha, it seems unexpected that the most recent version of `getstorybook` would install them rather than the latest stable release.

### Affected platforms 

I'm on OSX (10.13.3)

In addition, and I'm not sure if it is related, by I am getting the following error when installing Storybook (using `getstorybook -f`) and running `npm run storybook`:

```
module.js:559
    throw err;
    ^

Error: Cannot find module './Mime'
    at Function.Module._resolveFilename (module.js:557:15)
    at Function.Module._load (module.js:484:25)
    at Module.require (module.js:606:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/someguy/code/someproject/node_modules/mime/index.js:3:12)
    at Module._compile (module.js:662:30)
    at Object.Module._extensions..js (module.js:673:10)
    at Module.load (module.js:575:32)
    at tryModuleLoad (module.js:515:12)
    at Function.Module._load (module.js:507:3)
```

--- a/lib/cli/lib/has_yarn.js
+++ b/lib/cli/lib/has_yarn.js
@@ -1,10 +1,13 @@
+import { sync as spawnSync } from 'cross-spawn';
 import path from 'path';
 import fs from 'fs';
 
 export default function hasYarn() {
+  const yarnAvailable = spawnSync('yarn', ['--version'], { silent: true });
+  const npmAvailable = spawnSync('npm', ['--version'], { silent: true });
   const yarnLockPath = path.resolve('yarn.lock');
-  if (!fs.existsSync(yarnLockPath)) {
-    return false;
+  if ((fs.existsSync(yarnLockPath) || npmAvailable.status !== 0) && yarnAvailable.status === 0) {
+    return true;
   }
-  return true;
+  return false;
 }

--- a/lib/cli/lib/has_yarn.js
+++ b/lib/cli/lib/has_yarn.js
@@ -1,6 +1,10 @@
-import { sync as spawnSync } from 'cross-spawn';
+import path from 'path';
+import fs from 'fs';
 
 export default function hasYarn() {
-  const result = spawnSync('yarn', ['--version'], { silent: true });
-  return result.status === 0;
+  const yarnLockPath = path.resolve('yarn.lock');
+  if (!fs.existsSync(yarnLockPath)) {
+    return false;
+  }
+  return true;
 }

